src/Microsoft.AspNetCore.Authentication.Cookies/Events/BaseCookieContext.cs

```diff
@@ -23,5 +24,7 @@ public class BaseCookieContext : BaseContext
         }
 
         public CookieAuthenticationOptions Options { get; }
+
+        public AuthenticationScheme Scheme { get; }
```

Comment:
I tried to bridge the context's between the service and the handlers.  The service doesn't care about the scheme object, only the string, but maybe we should just have a BaseAuthenticationHandlerContext that does have the scheme.  I have a todo in the top comment to make a pass through all of our events/context objects since they have always been a bit of a hodgepodge mess